okio/src/commonMain/kotlin/okio/FileSystem.kt

```diff
@@ -145,7 +145,11 @@ expect abstract class FileSystem() {
    *     if there's a loop of symbolic links, or if any name is too long.
    */
   @Throws(IOException::class)
-  abstract fun open(file: Path): FileHandle
+  abstract fun open(
+    file: Path,
+    read: Boolean = false,
```

Comment:
Do you want @JvmOverloads for this?

Reply:
After spending more time with this I think there’s basically two modes:

openReadOnly – file must already exist, it is an error to attempt to write
openReadWrite – file will be created if it doesn’t exist

Thoughts on exposing this rather than having one function with boolean options?